base/strings/utf8proc.jl

```diff
@@ -74,8 +74,12 @@ function utf8proc_map(s::String, flags::Integer)
                    (Ptr{UInt8}, Cssize_t, Ref{Ptr{UInt8}}, Cint),
                    s, sizeof(s), p, flags)
     result < 0 && error(unsafe_string(ccall(:utf8proc_errmsg, Cstring,
-                                         (Cssize_t,), result)))
-    unsafe_wrap(String, p[], result, true)::String
+                                            (Cssize_t,), result)))
+    ptr = p[]
+    # TODO: avoid copy
```

Comment:
Yes.  You call the lower-level `utf8proc_decompose` and `utf8proc_reencode` functions as in the [utf8proc_map implementation](https://github.com/JuliaLang/utf8proc/blob/6271fb97c08e28b16fd39d440cb586c73c7d8a0f/utf8proc.c#L699-L727) and `utf8proc_decompose` will compute the size for you.